src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -668,12 +670,29 @@ private void TakeIncrementalSnapshot(BoundNode node)
             _snapshotBuilderOpt?.TakeIncrementalSnapshot(node, State);
         }
 
-        private void SetUpdatedSymbol(BoundNode node, Symbol originalSymbol, Symbol updatedSymbol)
+        private void SetLocalType(BoundNode sourceNode, LocalSymbol originalLocal, TypeWithAnnotations updatedType)
+        {
+            _variableTypes[originalLocal] = updatedType;
+
+            LocalSymbol updatedLocal;
+            if (_reinferredVariablesOpt is object)
+            {
+                updatedLocal = ((SourceLocalSymbol)originalLocal).WithContainingAndAnalyzedTypes(GetUpdatedContainingSymbol(originalLocal), updatedType);
+                if ((object)originalLocal == updatedLocal) return;
+                SetUpdatedSymbol(sourceNode, originalLocal, updatedLocal);
+                _reinferredVariablesOpt?.SetReinferredSymbol(originalLocal, updatedLocal);
+            }
+        }
+
+        private Symbol GetUpdatedContainingSymbol(Symbol childSymbol) =>
+            _reinferredVariablesOpt?.GetReinferredSymbol(childSymbol.ContainingSymbol) ?? childSymbol.ContainingSymbol;
+
+        private void SetUpdatedSymbol<T>(BoundNode node, T originalSymbol, T updatedSymbol) where T : Symbol
```

Comment:
><T> [](start = 37, length = 3)

Is the type parameter necessary?

Reply:
The type parameter is there to ensure that consumers pass the same original symbol and updated type at compile time. I messed this up once during development so I added harder guarantees.

---
In reply to: [327871357](https://github.com/dotnet/roslyn/pull/38769#discussion_r327871357) [](ancestors = 327871357)

Reply:
No longer relevant.

---
In reply to: [327873967](https://github.com/dotnet/roslyn/pull/38769#discussion_r327873967) [](ancestors = 327873967,327871357)